File: scripts/download-images.ts
import axios from 'axios';
import * as fs from 'fs';
import * as path from 'path';
import { Stream } from 'stream';

const games = [
  { name: 'pacman', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/pacman.jpg' },
  { name: 'arkanoid', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/arkanoid.jpg' },
  { name: 'hextris', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/hextris.jpg' },
  { name: 'duckhunt', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/duckhunt.jpg' },
  { name: 'hexgl', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/hexgl.jpg' },
  { name: 'lostworld', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/lostworld.jpg' },
  { name: 'powerline', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/powerline.jpg' },
  { name: 'diep', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/diep.jpg' },
  { name: 'wings', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/wings.jpg' },
  { name: 'starve', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/starve.jpg' },
  { name: 'gartic', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/gartic.jpg' },
  { name: 'territorial', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/territorial.jpg' },
  { name: 'deeeep', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/deeeep.jpg' },
  { name: 'splix', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/splix.jpg' },
  { name: 'generals', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/generals.jpg' },
  { name: 'slither', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/slither.jpg' },
  { name: 'curvefever', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/curvefever.jpg' },
  { name: 'evades', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/evades.jpg' },
  { name: 'brutal', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/brutal.jpg' },
  { name: 'wilds', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/wilds.jpg' },
  { name: 'stabfish', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/stabfish.jpg' },
  { name: 'krunker', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/krunker.jpg' },
  { name: 'shellshock', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/shellshock.jpg' },
  { name: 'zombsroyale', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/zombsroyale.jpg' },
  { name: 'surviv', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/surviv.jpg' },
  { name: '1v1lol', url: 'https://cdn.jsdelivr.net/gh/webgamesvip/webgamesvip.github.io/public/images/1v1lol.jpg' }
];

async function downloadImage(url: string, filepath: string) {
  const response = await axios({
    url,
    method: 'GET',
    responseType: 'stream'
  });

  return new Promise((resolve, reject) => {
    (response.data as Stream)
      .pipe(fs.createWriteStream(filepath))
      .on('finish', () => resolve('done'))
      .on('error', (e: Error) => reject(e));
  });
}

async function main() {
  const publicDir = path.join(process.cwd(), 'public');
  const gamesDir = path.join(publicDir, 'games');

  // Create directories if they don't exist
  if (!fs.existsSync(publicDir)) {
    fs.mkdirSync(publicDir);
  }
  if (!fs.existsSync(gamesDir)) {
    fs.mkdirSync(gamesDir);
  }

  for (const game of games) {
    const filepath = path.join(gamesDir, `${game.name}.jpg`);
    try {
      await downloadImage(game.url, filepath);
      console.log(`Downloaded ${game.name} image`);
    } catch (error) {
      console.error(`Error downloading ${game.name} image:`, error);
    }
  }
}

main(); 

File: scripts/download_images.py
import os
import requests
from PIL import Image, ImageDraw, ImageFont
import io

# 游戏列表
games = [
    {"title": "Pac-Man", "filename": "pacman.jpg", "search": "pacman arcade game"},
    {"title": "Arkanoid", "filename": "arkanoid.jpg", "search": "arkanoid breakout game"},
    {"title": "Snake", "filename": "snake.jpg", "search": "snake game classic"},
    {"title": "Hextris", "filename": "hextris.jpg", "search": "hextris game"},
    {"title": "Classic Minecraft", "filename": "minecraft.jpg", "search": "classic minecraft browser"},
    {"title": "Duck Hunt", "filename": "duckhunt.jpg", "search": "duck hunt nes game"},
    {"title": "Radius Raid", "filename": "radius-raid.jpg", "search": "radius raid game"},
    {"title": "HexGL", "filename": "hexgl.jpg", "search": "hexgl racing game"},
    {"title": "Lost World", "filename": "lostworld.jpg", "search": "lost world game"},
    {"title": "Powerline", "filename": "powerline.jpg", "search": "powerline io game"},
    {"title": "Zombs", "filename": "zombs.jpg", "search": "zombs io game"},
    {"title": "Diep", "filename": "diep.jpg", "search": "diep io game"},
    {"title": "Wings", "filename": "wings.jpg", "search": "wings io game"},
    {"title": "Starve", "filename": "starve.jpg", "search": "starve io game"},
    {"title": "2048", "filename": "2048.jpg", "search": "2048 puzzle game"},
    {"title": "Chess", "filename": "chess.jpg", "search": "chess game"},
    {"title": "Gartic", "filename": "gartic.jpg", "search": "gartic io game"},
    {"title": "Territorial", "filename": "territorial.jpg", "search": "territorial io game"},
    {"title": "Deeeep", "filename": "deeeep.jpg", "search": "deeeep io game"},
    {"title": "Splix", "filename": "splix.jpg", "search": "splix io game"},
    {"title": "Generals", "filename": "generals.jpg", "search": "generals io game"},
    {"title": "Slither", "filename": "slither.jpg", "search": "slither io game"},
    {"title": "Curve Fever", "filename": "curvefever.jpg", "search": "curve fever game"},
    {"title": "Evades", "filename": "evades.jpg", "search": "evades io game"},
    {"title": "Brutal", "filename": "brutal.jpg", "search": "brutal io game"},
    {"title": "Wilds", "filename": "wilds.jpg", "search": "wilds io game"},
    {"title": "Stab Fish", "filename": "stabfish.jpg", "search": "stabfish io game"},
    {"title": "Krunker", "filename": "krunker.jpg", "search": "krunker io game"},
    {"title": "Shell Shock", "filename": "shellshock.jpg", "search": "shell shock io game"},
    {"title": "Zombs Royale", "filename": "zombsroyale.jpg", "search": "zombs royale io game"},
    {"title": "Surviv", "filename": "surviv.jpg", "search": "surviv io game"},
    {"title": "1v1.LOL", "filename": "1v1lol.jpg", "search": "1v1 lol game"},
]

def create_placeholder_image(title, filename, size=(800, 600)):
    # 创建一个新的图片
    image = Image.new('RGB', size, color='#f0f0f0')
    draw = ImageDraw.Draw(image)
    
    # 绘制游戏标题
    title_text = title
    # 由于没有指定字体，这里使用简单的文本
    draw.text((size[0]/2, size[1]/2), title_text, fill='#666666', anchor="mm")
    
    # 保存图片
    image.save(f'public/games/{filename}')
    print(f"Created placeholder for {title}")

def main():
    # 确保目录存在
    os.makedirs('public/games', exist_ok=True)
    
    # 为每个游戏创建占位图片
    for game in games:
        try:
            create_placeholder_image(game['title'], game['filename'])
        except Exception as e:
            print(f"Error creating image for {game['title']}: {e}")

if __name__ == "__main__":
    main() 